### PR TITLE
[advertisement] Tick immediately on start

### PR DESCRIPTION
**Pull request by schallert**
When we start advertising it takes as long as the heartbeat interval
duration to begin advertising. Ideally a service shouldn't start
advertising until it's ready, so when it starts it might make sense to
show the advertisement immediately. Even if it's not ready at first, if
it has a health function it won't start heartbeating until it's ready
anyway.

RFR @cw9

### CODE REVIEW

File: services/client/services.go [hunk 1]
```diff
@@ -153,16 +153,26 @@ func (c *client) Advertise(ad services.Advertisement) error {
...
+
+		tickFn := func() {
+			if isHealthy(ad) {
+				if err := hb.Heartbeat(pi, m.LivenessInterval()); err != nil {
+					c.logger.Errorf("could not heartbeat service %s, %v", sid.String(), err)
+					errCounter.Inc(1)
+				}
+			}
+		}
+
```
Comment (cw9) [inline services/client/services.go]: just call tickFn() before the for loop?

File: services/client/services_test.go [hunk 2]
```diff
@@ -136,6 +136,16 @@ func TestAdvertiseErrors(t *testing.T) {
 	err = sd.Advertise(ad)
 	require.NoError(t, err)
 
+	hbGen.Lock()
+	hb := hbGen.hbs[serviceKey(sid)]
+	hbGen.Unlock()
+
+	// hb store should show advertisement (almost) immediately
+	time.Sleep(time.Millisecond)
```
Comment (cw9) [inline services/client/services_test.go]: can you try run this test with --count 1000? feeling this is gonna be flaky

Comment (cw9) [inline services/client/services_test.go]: the test case could be: construct the sd client with a super long HeartbeatInterval, like an hour, and verify the heartbeat still got created pretty soon

File: services/client/services.go (hunk 1 above)
Comment (schallert) [inline services/client/services.go]: Updated!

File: services/client/services_test.go (hunk 2 above)
Comment (schallert) [inline services/client/services_test.go]: Increased the relevant times to give the tests a bit more headroom. Ran with `-c 100` and didn't see any flakes so far.

**Pull request merged**